src/Type/Accessory/NonEmptyArrayType.php

```diff
@@ -354,6 +355,18 @@ public function isScalar(): TrinaryLogic
 
 	public function looseCompare(Type $type, PhpVersion $phpVersion): BooleanType
 	{
+		if ($type->isTrue()->yes()) {
+			return new ConstantBooleanType(true);
+		}
+
+		if ($type->isFalse()->yes() || $type->isNull()->yes()) {
```

Comment:
This kind of condition fails for `false|null`, because both left and right part are maybe. You need to use (new UnionType(false, null))->isSuperTypeOf($type)->yes()`.

You also need to include the below `if` in that union type too - an empty ConstantArrayType.

Reply:
fixed